packages/flutter/lib/src/cupertino/route.dart

```diff
@@ -152,8 +152,28 @@ mixin CupertinoRouteTransitionMixin<T> on PageRoute<T> {
 
   @override
   bool canTransitionTo(TransitionRoute<dynamic> nextRoute) {
```

Comment:
Overall I am wondering if this canTransitionTo/canTransitionFrom business still make sense? I thought at some point we had talked about giving the incoming route full power over the animation of the existing route? The duality that we have now seems kind of confusing.